doc/release/next_whats_new/radio_buttons_2d_grid.rst

```diff
@@ -0,0 +1,49 @@
+RadioButtons widget supports flexible layouts
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The `.widgets.RadioButtons` widget now supports arranging buttons in different
+layouts via the new *layout* parameter. You can arrange buttons vertically
+(default), horizontally, or in a 2D grid by passing a ``(rows, cols)`` tuple.
+
+The *active* parameter and the ``RadioButtons.index_selected`` attribute
+continue to use a single integer index into the labels list. For grid layouts,
+buttons are positioned left-to-right, top-to-bottom. The column positions are
+automatically calculated based on the maximum text width in each column,
+ensuring optimal spacing.
```

Comment:
```suggestion
(default), horizontally, or in a 2D grid.
```

This is too much detail for a what's new entry.

Not sure we need the "single integer index" info at all (IMHO this is more or less intuitive), but I you want to spell it out, the right place is the docstring.